### PR TITLE
Add unit tests for `Parallax2D`

### PR DESCRIPTION
**Pull request by LeonStansfield**
In this PR I have added unit tests for parallax_2d.h for the tracker #43440
The tests include
- set_scroll_scale and get_scroll_scale
- set_repeat_size and get_repeat_size
- set_repeat_times and get_repeat_times
- set_autoscroll and get_autoscroll
- set_scroll_offset and get_scroll_offset
- set_screen_offset and get_screen_offset
- set_limit_begin and get_limit_begin
- set_limit_end and get_limit_end
- set_follow_viewport and get_follow_viewport
- set_ignore_camera_scroll and is_ignore_camera_scroll

### CODE REVIEW

**Comment by Calinou** [comment]
cc @markdibarry

File: tests/scene/test_parallax_2d.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+#include "scene/2d/parallax_2d.h"
+
+#include "tests/test_macros.h"
+
+namespace TestParallax2D {
+
+TEST_CASE("[SceneTree][Parallax2D] Test Scroll Scale") {
+    MESSAGE("Scale test");
+    Parallax2D parallax;
```
Comment (AThousandShips) [inline tests/scene/test_parallax_2d.h:42]: These should all be created as objects with `memnew` I'd say, and then freed at the end of the test, generally Godot object classes need special initialization

**Comment by AThousandShips** [comment]
Please use clang-format or the precommit hook, see [here](https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html#using-clang-format-locally), to fix your style instead

**Review by markdibarry — APPROVED** [review]

**Comment by kleonc** [comment]
Please squash commits into one, see [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) (and ensure the resulting commit has a short descriptive title).

**Comment by AThousandShips** [comment]
Please fix the commit message to follow the style, the PR title would be good

File: tests/scene/test_parallax_2d.h [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	parallax->set_limit_end(limit_end);
+	CHECK(parallax->get_limit_end() == limit_end);
+	memdelete(parallax);
+}
+
+TEST_CASE("[SceneTree][Parallax2D] Test Follow Viewport") {
+	// Test setting and getting the follow viewport flag.
+	Parallax2D *parallax = memnew(Parallax2D);
+	parallax->set_follow_viewport(false);
+	CHECK(parallax->get_follow_viewport() == false);
```
Comment (AThousandShips) [inline tests/scene/test_parallax_2d.h:117]:
```suggestion
	CHECK_FALSE(parallax->get_follow_viewport());
```

File: tests/scene/test_parallax_2d.h [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	memdelete(parallax);
+}
+
+TEST_CASE("[SceneTree][Parallax2D] Test Ignore Camera Scroll") {
+	// Test setting and getting the ignore camera scroll flag.
+	Parallax2D *parallax = memnew(Parallax2D);
+	parallax->set_ignore_camera_scroll(true);
+	CHECK(parallax->is_ignore_camera_scroll() == true);
```
Comment (AThousandShips) [inline tests/scene/test_parallax_2d.h:125]:
```suggestion
	CHECK(parallax->is_ignore_camera_scroll());
```

**Review by AThousandShips — APPROVED** [review]
Style looks good otherwise

File: tests/scene/test_parallax_2d.h [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+#define TEST_PARALLAX_2D_H
+
+#include "scene/2d/parallax_2d.h"
+#include "tests/test_macros.h"
+
+namespace TestParallax2D {
+
+// Test cases for the Parallax2D class to ensure its properties are set and retrieved correctly.
+
+TEST_CASE("[SceneTree][Parallax2D] Test Scroll Scale") {
```
Comment (Calinou) [inline tests/scene/test_parallax_2d.h:41]: Remove "Test" from all test case names, as it's redundant:

```suggestion
TEST_CASE("[SceneTree][Parallax2D] Scroll Scale") {
```

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**